core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -1036,6 +1031,44 @@ sub_module_instantiate(WASMModule *module, WASMModuleInstance *module_inst,
         (void)ret;
 
         sub_module_list_node = bh_list_elem_next(sub_module_list_node);
+
+#if WASM_ENABLE_LIBC_WASI != 0
+        {
+            /*
+             * reactor instances may assume that _initialize will be called by
+             * the environment at most once, and that none of their other
+             * exports are accessed before that call.
+             */
+            WASMFunctionInstance *initialize =
+                wasm_lookup_function(sub_module_inst, "_initialize", NULL);
+            /* a strong restricttion that a reactor must export a _initialize
```

Comment:
Same as above, seems that it doesn't require that `_initialize` function must be exported.

Reply:
And we should not use macro WASM_ENABLE_LIBC_WASI to check that the wasm app is in wasi mode or not, normally we should use `module->is_wasi_mode`

Reply:
Well, it is what I am worried about. `is_wasi_mode` doesn't always equal to `WASM_ENABLE_LIBC_WASI=1`. It represents a module depends on WASI APIs. If a reactor compiled by WASI_SDK and doesn't include any standard libraries, it will not have any imported entries but still a reactor.

``` c
void foo() {}
```

Reply:
Yes, it is still a reactor and exports _inititialize, but the function does nothing, whehter to call it or not seems meaningless. There ight be two choices: (1) Call _initialize it is found, (2) Call _initialize when it is found and the module is wasi-module.

And also we should check the func signature of `_start` and `_initialize`, they must be "()".

Reply:
I tend to make it simple without considering .wasm is import any `wasi_snapshot_preview1` function or not.